packages/rest/src/lib/RequestManager.ts

```diff
@@ -181,6 +191,43 @@ export class RequestManager extends EventEmitter {
 		this.options = { ...DefaultRestOptions, ...options };
 		this.options.offset = Math.max(0, this.options.offset);
 		this.globalRemaining = this.options.globalRequestsPerSecond;
+
+		setInterval(() => {
+			// Only allocate a swept collection if there are listeners
+			let sweptHashes: Collection<string, HashData> | null = null;
+			const isListeningToSweeps = this.listenerCount(RESTEvents.HashSweep) > 0;
+			if (isListeningToSweeps) {
+				sweptHashes = new Collection<string, HashData>();
+			}
+
+			const curDate = Date.now();
+
+			// Begin sweep hash based on lifetimes
+			this.hashes.sweep((v, k) => {
+				// `-1` indicates a global hash
+				if (v.lastAccess === -1) return false;
+
+				// Check if lifetime has been exceeded
+				const shouldSweep = Math.floor(curDate - v.lastAccess) > this.options.hashLifetime;
+
+				// Add hash to collection of swept hashes
+				if (shouldSweep && isListeningToSweeps) {
```

Comment:
You should always emit the debug event if needed, even if not listening to sweeps